src/main/java/org/mifos/identityaccountmapper/service/AccountLookupService.java

```diff
@@ -0,0 +1,55 @@
+package org.mifos.identityaccountmapper.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.mifos.identityaccountmapper.data.BeneficiaryDTO;
+import org.mifos.identityaccountmapper.domain.IdentityDetails;
+import org.mifos.identityaccountmapper.domain.PaymentModalityDetails;
+import org.mifos.identityaccountmapper.repository.ErrorTrackingRepository;
+import org.mifos.identityaccountmapper.repository.MasterRepository;
+import org.mifos.identityaccountmapper.repository.PaymentModalityRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AccountLookupService {
+    private final MasterRepository masterRepository;
+    private final ErrorTrackingRepository errorTrackingRepository;
+    private final PaymentModalityRepository paymentModalityRepository;
+    private final SendCallbackService sendCallbackService;
+    private final ObjectMapper objectMapper;
+    private static final Logger logger = LoggerFactory.getLogger(AccountLookupService.class);
+
+    @Autowired
+    public AccountLookupService(MasterRepository masterRepository, ErrorTrackingRepository errorTrackingRepository,
+                                           PaymentModalityRepository paymentModalityRepository, SendCallbackService sendCallbackService,
+                                ObjectMapper objectMapper){
+        this.masterRepository = masterRepository;
+        this.errorTrackingRepository = errorTrackingRepository;
+        this.paymentModalityRepository = paymentModalityRepository;
+        this.sendCallbackService = sendCallbackService;
+        this.objectMapper = objectMapper;
+    }
+    @Async("asyncExecutor")
+    public void accountLookup(String callbackURL,String payeeIdentity){
+        try {
+            IdentityDetails identityDetails = masterRepository.findByPayeeIdentity(payeeIdentity).orElseThrow(()-> new RuntimeException("Payee Identity does not exist."));
```

Comment:
create this as a custom exception` new RuntimeException("Payee Identity does not exist.")`

Reply:
Created new PayeeIdentityException